src/main/java/graphql/execution/FieldCollector.java

```diff
@@ -52,13 +53,13 @@ public MergedSelectionSet collectFields(FieldCollectorParameters parameters, Mer
      */
     public MergedSelectionSet collectFields(FieldCollectorParameters parameters, SelectionSet selectionSet) {
         Map<String, MergedField> subFields = new LinkedHashMap<>();
-        List<String> visitedFragments = new ArrayList<>();
+        Set<String> visitedFragments = new HashSet<>();
```

Comment:
same as above: We should use LinkedHashSet here to be predictable and make debugging easier.